### PR TITLE
Pipeline resource limits

### PR DESCRIPTION
**Pull request by sjezewski**
This PR:

- renames `resource_spec` to `resource_requests` in the pipeline spec
- adds support for `resource_limits` in the pipeline spec
- uses the limits to specify GPU resources

### CODE REVIEW

**Comment by JoeyZwicker** [comment]
Please make sure you update the pipeline spec docs as well as open issues for broader documentation that may need to be done.

File: doc/reference/pipeline_spec.md [hunk 1]
```diff
@@ -31,7 +31,11 @@ create-pipeline](../pachctl/pachctl_create-pipeline.html) doc.
     "constant": int
     "coefficient": double
   },
-  "resource_spec": {
+  "resource_requests": {
+    "memory": string
```
Comment (jdoliner) [inline doc/reference/pipeline_spec.md]: I think there should be a comma here, although it looks like it was already missing.

**Comment by jdoliner** [comment]
Actually 1 very minor comment about a missing comma in the docs.

File: doc/reference/pipeline_spec.md (hunk 1 above)
Comment (sjezewski) [reply doc/reference/pipeline_spec.md]: Ok ... I was trying to figure out what was wrong ... didn't realize GH was smart enough to do syntax linting on json code snippets ... thats nifty

**Comment by sjezewski** [comment]
Additionally, I've tested this on k8s 1.8 and it gets us past the GPU error we were seeing.

File: .travis.yml [hunk 2]
```diff
@@ -18,6 +18,8 @@ before_script:
 script:
 - make lint vet
 - ./etc/testing/travis.sh
+after_failure:
```
Comment (sjezewski) [inline .travis.yml]: This was a bit more idiomatic ... we want this info when tests fail. Was helpful during my debugging.

Comment (jdoliner) [reply .travis.yml]: :+1:

**Pull request merged**